micro-rdk-ffi/src/ffi/config.rs

```diff
@@ -135,3 +135,70 @@ pub unsafe extern "C" fn config_get_i32(
     unsafe { *out = val };
     errors::viam_code::VIAM_OK
 }
+
+/// Get a vector of int32s from the attribute section of a sensor configuration,
+/// if found the values will be copied into `out`. The length should be obtained
+/// first, using `config_get_i32_vec_len` in order to allocate the proper amount of memory
+/// pointed to by `out`
+///
+/// # Safety
+/// `ctx`, `key`, `out` must be valid pointers for the duration of the call
+/// `key` must be a null terminated C string. Additionally the external process calling
+/// the function is responsible for managing the memory allocated for `out`
+#[no_mangle]
+pub unsafe extern "C" fn config_get_i32_vec(
```

Comment:
Did you consider returning a rust allocated array instead? so be copied in C and returned or kept until not needed anymore?

Reply:
I did but I thought having the Rust code write to memory pre-allocated by C was safer and simpler? I might not have approached the Rust allocated array properly when I attempted it